src/pfe/portal/modules/User.js

```diff
@@ -736,30 +725,18 @@ module.exports = class User {
       log.info("The Docker config file has been updated for " + url);
     } catch (err) {
       const msg = "Failed to update the Codewind Docker Config File";
-      log.error(err);
       log.error(msg);
-      const retval = {
-        statusCode: 500,
-        body: msg
-      };
-      return retval;
+      throw err;
     }
 
     // Only update the Kube secret when running in K8s
     let isServiceAccountPatched = false;
     if (global.codewind.RUNNING_IN_K8S) {
       isServiceAccountPatched = await this.updateServiceAccountWithDockerRegisrySecret();
-    } else {
-      // we are in local Codewind and we just return
-      log.debug("Codewind Docker Registry List: " + JSON.stringify(registrySecretList));
-      const retval = {
-        statusCode: 201,
-        body: registrySecretList
-      };
-      return retval;
     }
 
-    if (isServiceAccountPatched) {
+    if (isServiceAccountPatched || !global.codewind.RUNNING_IN_K8S) {
+      // when either the service account is patched successfully or we are in local Codewind, we can return
       log.debug("Codewind Docker Registry List: " + JSON.stringify(registrySecretList));
       const retval = {
         statusCode: 201,
```

Comment:
I think you can just return the list here. The `POST /api/v1/registrysecrets` route can decide that's a 201 response. That way any other part of pfe that wants the list of registry secrets can call this function and get a list. (And this function doesn't need to know anything about http response codes.)

Reply:
updated